More significant warnings on CRAN R-devel

After submitting to CRAN, it came up with more significant warnings on one platform:

```
* using log directory ‘/srv/hornik/tmp/CRAN/Rttf2pt1.Rcheck’
* using R Under development (unstable) (2022-10-05 r83033)
* using platform: x86_64-pc-linux-gnu (64-bit)
* using session charset: UTF-8
* checking for file ‘Rttf2pt1/DESCRIPTION’ ... OK
* this is package ‘Rttf2pt1’ version ‘1.3.11’
* package encoding: UTF-8
* checking CRAN incoming feasibility ... [2s/2s] NOTE
Maintainer: ‘Winston Chang <winston@stdout.org>’

CRAN repository db overrides:
  License_is_FOSS: yes
* checking package namespace information ... OK
* checking package dependencies ... OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for executable files ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking for sufficient/correct file permissions ... OK
* checking serialization versions ... OK
* checking whether package ‘Rttf2pt1’ can be installed ... [5s/5s] WARNING
Found the following significant warnings:
  ttf2pt1.c:659:41: warning: format ‘%i’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  ttf2pt1.c:688:47: warning: format ‘%i’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  ttf2pt1.c:703:67: warning: format ‘%i’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  ttf2pt1.c:720:48: warning: format ‘%i’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  ttf2pt1.c:1081:36: warning: format ‘%d’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  t1asm.c:173:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  t1asm.c:250:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  t1asm.c:270:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  t1asm.c:295:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  t1asm.c:309:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  t1asm.c:357:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  t1asm.c:384:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  t1asm.c:428:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  t1asm.c:463:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  t1asm.c:476:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
See ‘/srv/hornik/tmp/CRAN/Rttf2pt1.Rcheck/00install.out’ for details.
```

More info extracted from 00install.out:


```
ttf2pt1.c: In function ‘unicode_init_user’:
ttf2pt1.c:659:41: warning: format ‘%i’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  659 |                 if( sscanf(buffer, "at %i", &curpos) == 1 ) {
      |                                        ~^   ~~~~~~~
      |                                         |   |
      |                                         |   unsigned int *
      |                                         int *
      |                                        %i
ttf2pt1.c:688:47: warning: format ‘%i’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  688 |                 else if( (n=sscanf(buffer, " %i%n", &unicode, &cnt)) == 1 ) {
      |                                              ~^     ~~~~~~~~
      |                                               |     |
      |                                               int * unsigned int *
      |                                              %i
ttf2pt1.c:703:67: warning: format ‘%i’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  703 |                                                 if ( sscanf(p, " %i%n", &unicode2, &cnt) != 1 ) {
      |                                                                  ~^     ~~~~~~~~~
      |                                                                   |     |
      |                                                                   int * unsigned int *
      |                                                                  %i
ttf2pt1.c:720:48: warning: format ‘%i’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
  720 |                         } while ( sscanf(p, " %i%n", &unicode, &cnt) == 1 );
      |                                               ~^     ~~~~~~~~
      |                                                |     |
      |                                                int * unsigned int *
      |                                               %i
ttf2pt1.c: In function ‘unicode_plane’:
ttf2pt1.c:1081:36: warning: format ‘%d’ expects argument of type ‘int *’, but argument 3 has type ‘unsigned int *’ [-Wformat=]
 1081 |                 c2 = sscanf(arg, "%d", &plane);
      |                                   ~^   ~~~~~~
      |                                    |   |
      |                                    |   unsigned int *
      |                                    int *
      |                                   %d


t1asm.c:173:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  173 | static void output_block()
      |             ^~~~~~~~~~~~
t1asm.c:250:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  250 | static void eexec_start()
      |             ^~~~~~~~~~~
t1asm.c:270:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  270 | static void eexec_end()
      |             ^~~~~~~~~
t1asm.c:295:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  295 | static void file_end()
      |             ^~~~~~~~
t1asm.c:309:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  309 | static void t1asm_getline()
      |             ^~~~~~~~~~~~~
t1asm.c:357:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  357 | static void charstring_start()
      |             ^~~~~~~~~~~~~~~~
t1asm.c:384:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  384 | static void charstring_end()
      |             ^~~~~~~~~~~~~~
t1asm.c:428:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  428 | static void parse_charstring()
      |             ^~~~~~~~~~~~~~~~
t1asm.c:463:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  463 | static void usage()
      |             ^~~~~
t1asm.c:476:13: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  476 | static void print_banner()
      |             ^~~~~~~~~~~~
```

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,7 +1,7 @@
 Version 1.3.10.9000
 -------------
 
-* Closed #15: fixed `-Wformat` and `-Wformat-pedantic` warnings, which raise significant warnings on R-devel. (#16)
+* Closed #15, #17: fixed `-Wformat` and `-Wformat-pedantic` warnings, which raise significant warnings on R-devel. (#16, #18)
 
 Version 1.3.10
 -------------

--- a/src/ttf2pt1/t1asm.c
+++ b/src/ttf2pt1/t1asm.c
@@ -170,7 +170,7 @@ static byte cencrypt(byte plain)
 
 /* This function flushes a buffered PFB block. */
 
-static void output_block()
+static void output_block(void)
 {
   int32 i;
 
@@ -247,7 +247,7 @@ static void eexec_string(char *string)
    PFB format then flush current ASCII block and get ready for binary block.
    We start encryption with four random (zero) bytes. */
 
-static void eexec_start()
+static void eexec_start(void)
 {
   eexec_string(line);
   if (pfb) {
@@ -267,7 +267,7 @@ static void eexec_start()
    If output is in PFB format then this entails flushing binary block and
    starting an ASCII block. */
 
-static void eexec_end()
+static void eexec_end(void)
 {
   int i, j;
 
@@ -292,7 +292,7 @@ static void eexec_end()
    If output is in PFB format then this entails flushing binary block and
    starting an ASCII block. */
 
-static void file_end()
+static void file_end(void)
 {
   if (pfb) {
     output_block();
@@ -306,7 +306,7 @@ static void file_end()
    the newline is put into line[].  When terminated by '{', the '{' is not put
    into line[], and the flag start_charstring is set to 1. */
 
-static void t1asm_getline()
+static void t1asm_getline(void)
 {
   int c;
   char *p = line;
@@ -354,7 +354,7 @@ static int is_integer(char *string)
 /* This function initializes charstring encryption.  Note that this is called
    at the beginning of every charstring. */
 
-static void charstring_start()
+static void charstring_start(void)
 {
   int i;
 
@@ -381,7 +381,7 @@ static void charstring_byte(int v)
 /* This function outputs buffered, encrypted charstring data through possible
    eexec encryption. */
 
-static void charstring_end()
+static void charstring_end(void)
 {
   byte *bp;
 
@@ -425,7 +425,7 @@ static void charstring_int(int num)
 /* This function parses an entire charstring into integers and commands,
    outputting bytes through the charstring buffer. */
 
-static void parse_charstring()
+static void parse_charstring(void)
 {
   struct command *cp;
 
@@ -460,7 +460,7 @@ static void parse_charstring()
   charstring_end();
 }
 
-static void usage()
+static void usage(void)
 {
   fprintf(stderr,
           "usage: t1asm [-b] [-l block-length] [input [output]]\n");
@@ -473,7 +473,7 @@ static void usage()
   exit(1);
 }
 
-static void print_banner()
+static void print_banner(void)
 {
   static char rcs_revision[] = ""; /* removed RCS */
   static char revision[20];

--- a/src/ttf2pt1/ttf2pt1.c
+++ b/src/ttf2pt1/ttf2pt1.c
@@ -656,7 +656,7 @@ unicode_init_user(
 		if( !enabled )
 			continue; /* skip to the next plane */
 
-		if( sscanf(buffer, "at %i", &curpos) == 1 ) {
+		if( sscanf(buffer, "at %i", (int*)&curpos) == 1 ) {
 			if(curpos > 255) {
 				fprintf(stderr, "**** map file '%s' line %d: code over 255\n", path, lineno);
 				exit(1);
@@ -685,7 +685,7 @@ unicode_init_user(
 			}
 		}
 		/* try the compact sequence format */
-		else if( (n=sscanf(buffer, " %i%n", &unicode, &cnt)) == 1 ) {
+		else if( (n=sscanf(buffer, " %i%n", (int*)&unicode, &cnt)) == 1 ) {
 			p = buffer;
 			do {
 				if(curpos > 255) {
@@ -700,7 +700,7 @@ unicode_init_user(
 					if(ISDBG(EXTMAP)) fprintf(stderr, "=== next: '%c'\n", next);
 					p += cnt;
 					if( next == '-' ) { /* range */
-						if ( sscanf(p, " %i%n", &unicode2, &cnt) != 1 ) {
+						if ( sscanf(p, " %i%n", (int*)&unicode2, &cnt) != 1 ) {
 							fprintf(stderr, "**** map file '%s' line %d: missing end of range\n", path, lineno);
 							exit(1);
 						}
@@ -717,7 +717,7 @@ unicode_init_user(
 						}
 					}
 				}
-			} while ( sscanf(p, " %i%n", &unicode, &cnt) == 1 );
+			} while ( sscanf(p, " %i%n", (int*)&unicode, &cnt) == 1 );
 		}
 
 	}
@@ -1078,7 +1078,7 @@ unicode_plane(
 		arg += 2;
 		c2 = sscanf(arg, "%x", &plane);
 	} else {
-		c2 = sscanf(arg, "%d", &plane);
+		c2 = sscanf(arg, "%u", &plane);
 	}
 
 	if( (c1!=2 && c1!=0) || (c1==0 && c2==0) ) {

